### PR TITLE
remove repeated gradient step

### PR DESCRIPTION
**Pull request by OscarXZQ**

### CODE REVIEW

File: rnn/models/lstm_oneflow.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import oneflow.experimental as flow
+import oneflow.experimental.nn as nn
+import random
+import math
+# from torch.jit import Tensor  # there is an error
```
Comment (Ldpe2G) [inline rnn/models/lstm_oneflow.py:5]: 没用的代码就删除吧

File: rnn/models/lstm_oneflow.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        self.batch_size = batch_size
+        self.num_layers = num_layers
+ 
+        # Define the LSTM layer
+        self.lstm = CustomLSTM(self.input_dim, self.hidden_dim)
+ 
+        # Define the output layer
+        self.linear = nn.Linear(self.hidden_dim, output_dim)
+ 
+    # def initHidden(self):
```
Comment (Ldpe2G) [inline rnn/models/lstm_oneflow.py:22]: 这段代码如果没用到就可以删掉了

File: rnn/models/lstm_oneflow.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    def forward(self, input):
+        # Forward pass through LSTM layer
+        # shape of lstm_out: [input_size, batch_size, hidden_dim]
+        # shape of self.hidden: (a, b), where a and b both 
+        # have shape (num_layers, batch_size, hidden_dim).
+        lstm_out, _ = self.lstm(input.reshape((input.shape[0], self.batch_size, -1)))
+        
+        # Only take the output from the final timetep
+        # Can pass on the entirety of lstm_out to the next layer if it is a seq2seq prediction
+        #NOTE(Xu Zhiqiu) Negative indexing not supported
```
Comment (Ldpe2G) [inline rnn/models/lstm_oneflow.py:36]: # 号后面没有空格

File: rnn/train_lstm_oneflow.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+flow.enable_eager_execution()
+
+def _parse_args():
+    parser = argparse.ArgumentParser("flags for compare oneflow and pytorch speed")
+    parser.add_argument('--seed', nargs='?', type=int, const=0, help='specify random seed')
+    return parser.parse_args()
+
+def train(category_tensor, line_tensor, rnn, criterion, of_sgd):
+    output = rnn(line_tensor)
+    #print(output, category_tensor)
```
Comment (Ldpe2G) [inline rnn/train_lstm_oneflow.py]: 没有的注释代码都删了

File: rnn/train_rnn_oneflow.py [hunk 5]
```diff
@@ -23,6 +23,7 @@ def train(category_tensor, line_tensor, rnn, criterion, of_sgd):
     hidden = rnn.initHidden()
     for i in range(line_tensor.size()[0]):
         output, hidden = rnn(line_tensor[i], hidden)
+    #print(output, category_tensor)
```
Comment (Ldpe2G) [inline rnn/train_rnn_oneflow.py:26]: 删除

File: rnn/models/lstm_oneflow.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
+import oneflow.experimental as flow
+import oneflow.experimental.nn as nn
+import math
+
+#Reference: https://github.com/piEsposito/pytorch-lstm-by-hand
```
Comment (Ldpe2G) [inline rnn/models/lstm_oneflow.py:5]: \# 号后空格

File: rnn/models/lstm_oneflow.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        self.linear = nn.Linear(self.hidden_dim, output_dim)
+
+    def forward(self, input):
+        # Forward pass through LSTM layer
+        # shape of lstm_out: [input_size, batch_size, hidden_dim]
+        # shape of self.hidden: (a, b), where a and b both 
+        # have shape (num_layers, batch_size, hidden_dim).
+        lstm_out, _ = self.lstm(input.reshape((input.shape[0], self.batch_size, -1)))
+        
+        # Only take the output from the final timetep
```
Comment (Ldpe2G) [inline rnn/models/lstm_oneflow.py:28]: timetep -> timestep

**Review by Ldpe2G — APPROVED** [review]
